SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRumBuilder.swift

```diff
@@ -0,0 +1,144 @@
+//
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+
+@objc public class SplunkRumBuilder: NSObject {
+
+    private var beaconUrl: String
+    private var rumAuth: String
+    private var allowInsecureBeacon: Bool = false
+    private var debug: Bool = false
+    private var globalAttributes: [String: Any] = [:]
+    private var environment: String?
+    private var ignoreURLs: NSRegularExpression?
+    private var screenNameSpans: Bool = true
+    private var networkInstrumentation: Bool = true
+    private var enableDiskCache: Bool = false
+    private var spanDiskCacheMaxSize: Int64 = DEFAULT_DISK_CACHE_MAX_SIZE_BYTES
+    private var slowFrameDetectionThresholdMs: Double = 16.7
+    private var frozenFrameDetectionThresholdMs: Double = 700
+    private var sessionSamplingRatio: Double = 1.0
+
+    public init(beaconUrl: String, rumAuth: String) {
+        self.beaconUrl = beaconUrl
+        self.rumAuth = rumAuth
+    }
+
+    @discardableResult
+    @objc
+    public func allowInsecureBeacon(enabled: Bool) -> SplunkRumBuilder {
+        self.allowInsecureBeacon = enabled
+        return self
+    }
+
+    @discardableResult
+    @objc
+    public func debugEnabled(enabled: Bool) -> SplunkRumBuilder {
```

Comment:
Maybe a question of personal taste, but perhaps it would make sense to skip the double `enabled` wording by omitting the parameter name? I.e. `.debugEnabled(enabled: true)` vs `.debugEnabled(true)`

Reply:
I am not sure that an unnamed parameter would work here since it needs to be used in the enclosed function.  I can change the signature to "debug(enabled: Bool)" to shorten the name and remove the duplicate "enabled"